### PR TITLE
fix: ChooseLUN uses next highest lun before wrapping

### PR DESCRIPTION
**Pull request by jskazinski**
Change ChooseLUN to pick the next highest available lun before wrapping. This avoids an issue where the lun is reused and the determine multipath code in the iscsi lib get confused because the file system is very busy trying to catch up with all the new devices. This allows the pvc-hell stress test to pass but will need to research an alternate approach such as using WWN.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -330,6 +330,11 @@ func (client *Client) chooseLUN(initiatorName string) (int, error) {
 
 	sort.Sort(Volumes(volumes))
 
+	klog.V(5).Infof("checking if next LUN is not above maximum LUNs limit")
+	if volumes[len(volumes)-1].LUN+1 < MaximumLUN {
```
Comment (David-T-White) [inline api.go:334]: Looks like this could throw an error for a negative slice index if len(volumes) is 0, checked in the next block

Comment (jskazinski) [reply api.go:339]: Yes, good catch. Corrected to handle a volume slice of zero length.

**Review by David-T-White — APPROVED** [review]
looks good!

**Review by seagate-chris — APPROVED** [review]
Looks good if Dave's comment is addressed.

**Pull request merged**